### PR TITLE
KTOR-4593 Add support to automatically use the correct translation file for Pebble based on Accept Language header

### PR DESCRIPTION
**Pull request by TreyBastian**
**Subsystem**
Ktor-Server-Pebble

**Motivation**
This PR adds support for improving the ergonomics of using pebbles built in I18N support. It essentially makes it so you do not need to match Accept Headers to Locales in every route.

**Solution**
This uses a similar approach as the #1632 using `call.request.acceptLanguages` and validating it against a list of supported translations to either set the Locale or keep it `null` which is the default state.

### CODE REVIEW

**Comment by TreyBastian** [comment]
What would the best way to update the docs for this be for once this is merged? There would be code snippets so would it be after a release?

File: ktor-server/ktor-server-plugins/ktor-server-pebble/jvm/src/io/ktor/server/pebble/Pebble.kt
Comment (rsinukov) [inline ktor-server/ktor-server-plugins/ktor-server-pebble/jvm/src/io/ktor/server/pebble/Pebble.kt:18]: Can you replace an array with a mutable list, please?

File: ktor-server/ktor-server-plugins/ktor-server-pebble/jvm/src/io/ktor/server/pebble/Pebble.kt
Comment (rsinukov) [inline ktor-server/ktor-server-plugins/ktor-server-pebble/jvm/src/io/ktor/server/pebble/Pebble.kt:54]: can you extract locale to `if/else` block and this line outside of it?

File: ktor-server/ktor-server-plugins/ktor-server-pebble/jvm/src/io/ktor/server/pebble/Pebble.kt
Comment (rsinukov) [reply ktor-server/ktor-server-plugins/ktor-server-pebble/jvm/src/io/ktor/server/pebble/Pebble.kt:18]: Please add an API doc

**Review by rsinukov — CHANGES_REQUESTED** [review]
Thanks for the PR.
It's almost good to go, but minor comments. 
Regarding the docs, please create a ticket for this change in YouTrack, and we will make sure that it has proper docs after it's merged.

**Comment by TreyBastian** [comment]
> 

Thanks for reviewing. I made the changes I'll submit the YouTrack ticket in the morning. Let me know if there's anything else that should change. I've tried to stick with conventions I've seen in other plugins.

**Comment by TreyBastian** [comment]
https://youtrack.jetbrains.com/issue/KTOR-4593/Allow-Pebble-to-use-Accepted-Language-header-for-built-in-i18n-support

File: ktor-server/ktor-server-plugins/ktor-server-pebble/jvm/src/io/ktor/server/pebble/Pebble.kt
Comment (rsinukov) [inline ktor-server/ktor-server-plugins/ktor-server-pebble/jvm/src/io/ktor/server/pebble/Pebble.kt:26]: Actually, it can be just `List` and not `MutableList`. Sorry =)

File: ktor-server/ktor-server-plugins/ktor-server-pebble/jvm/src/io/ktor/server/pebble/Pebble.kt
Comment (rsinukov) [inline ktor-server/ktor-server-plugins/ktor-server-pebble/jvm/src/io/ktor/server/pebble/Pebble.kt:56]: please extract it to `createApplicationPlugin` to save the reference at install time to ignore possible mutations of config later

**Review by rsinukov — CHANGES_REQUESTED** [review]
Couple more changes.

**Comment by TreyBastian** [comment]
Sorry for the delay there those changes are in.

**Review by rsinukov — APPROVED** [review]
LGTM!
Will merge after CI run

**Pull request merged**
